hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -99,7 +99,7 @@ def required_args(target, args=(), kwargs=()):
     """
     try:
         spec = getfullargspec(
-            target.__init__ if inspect.isclass(target) else target)
+            target.__init__ if inspect.isclass(target) and '__init__' in target.__dict__ else target)
```

Comment:
This line has become complex enough that I'd split it out into explicit cases:

```python
if inspect.isclass(target) and hasattr(target, '__init__'):
    spec = getfullargspec(target.__init__)
else:
    spec = getfullargspec(target)
```
(this will also fix the lint errors you're seeing)

Reply:
looks good. done